soc/xtensa/esp32/linker.ld

```diff
@@ -186,67 +221,52 @@ SECTIONS
   Z_ITERABLE_SECTION_RAM(_static_thread_data, 4)
 
 #pragma push_macro("Z_ITERABLE_SECTION_RAM")
+#pragma push_macro("Z_ITERABLE_SECTION_RAM_GC_ALLOWED")
+#undef Z_ITERABLE_SECTION_RAM_GC_ALLOWED
+#define Z_ITERABLE_SECTION_RAM_GC_ALLOWED(x, y)
 #undef Z_ITERABLE_SECTION_RAM
 #define Z_ITERABLE_SECTION_RAM(x, y)
 #include <linker/common-ram.ld>
+/* Restore original value for symbols referenced by `common-ram.ld` */
+_net_buf_pool_list = _esp_net_buf_pool_list;
+#pragma pop_macro("Z_ITERABLE_SECTION_RAM_GC_ALLOWED")
 #pragma pop_macro("Z_ITERABLE_SECTION_RAM")
 
-  SECTION_DATA_PROLOGUE(log_static_section,,)
+  SECTION_PROLOGUE(_RODATA_SECTION_NAME,,ALIGN(20))
   {
-    __log_const_start = .;
+    _rodata_start = ABSOLUTE(.);
+
+#if defined(CONFIG_NET_SOCKETS)
+    . = ALIGN(4);
+    Z_LINK_ITERABLE(net_socket_register);
+    . = ALIGN(4);
+#endif
+
+#if defined(CONFIG_SETTINGS)
+    . = ALIGN(4);
+    Z_LINK_ITERABLE(settings_handler_static);
+    . = ALIGN(4);
+#endif
+
+    . = ALIGN(4);
+    __esp_log_const_start = .;
     KEEP(*(SORT(.log_const_*)));
-    __log_const_end = .;
+    __esp_log_const_end = .;
 
-    __log_backends_start = .;
+    . = ALIGN(4);
+    __esp_log_backends_start = .;
```

Comment:
I don't think this change is correct.  The logging backend is looking for the unprefixed symbols.